### ISSUE TITLE
*SOLVED* Getting Error when Running - Log Included

### ISSUE THREAD

**Issue by Heartbreaker714** [tr4wzified/KnowYourArmorPatcher#21]
Hate to say it but I think I'm getting the same (or similar) error as the other post. I deleted the _user/admin/app data/synthesis_ file and didn't help. But maybe I missed something...
<details>
  <summary>Know Your Armor - Error Log:</summary>
  
  Compiling
Extra data folder already exists. Leaving as is: "C:\Games\Phoenix Flavour\tools\Synthesis\Data\Skyrim Special Edition\KnowYourArmorPatcher"
("C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Git\pgzadd3g.e31\Runner\KnowYourArmorPatcher"): "dotnet" "build \"KnowYourArmorPatcher.csproj\" --runtime win-x64 -c Release"
Compiled
Sha "23cded3450c4bd0dfb9ff6bd1242eeadd975b739"
(""): "dotnet" "run --project \"C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Git\pgzadd3g.e31\Runner\KnowYourArmorPatcher\KnowYourArmorPatcher.csproj\" --runtime win-x64 -c Release --no-build check-runnability --DataFolderPath \"C:\Games\STEAM\steamapps\common\Skyrim Special Edition\Data\" --GameRelease SkyrimSE --LoadOrderFilePath C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Workspace\Plugins.txt"
Running
(""): "dotnet" "run --project \"C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Git\pgzadd3g.e31\Runner\KnowYourArmorPatcher\KnowYourArmorPatcher.csproj\" --runtime win-x64 -c Release --no-build run-patcher --DataFolderPath \"C:\Games\STEAM\steamapps\common\Skyrim Special Edition\Data\" --ExtraDataFolder \"C:\Games\Phoenix Flavour\tools\Synthesis\Data\Skyrim Special Edition\KnowYourArmorPatcher\" --DefaultDataFolderPath C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Git\pgzadd3g.e31\Runner\KnowYourArmorPatcher\Data --GameRelease SkyrimSE --LoadOrderFilePath C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Workspace\Plugins.txt --OutputPath \"C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Workspace\1 - KnowYourArmorPatcher\" --PatcherName KnowYourArmorPatcher --PersistencePath C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Persistence"
Mutagen version: 0.33.2
Mutagen sha: f14bb1e2d9591e1f23e4785686fb22db7c22365a
Synthesis version: 0.20.4
Synthesis sha: c2f7d7adcde3c39ad375b6ef1053e3b6836609fb
run-patcher --DataFolderPath "C:\Games\STEAM\steamapps\common\Skyrim Special Edition\Data" --ExtraDataFolder "C:\Games\Phoenix Flavour\tools\Synthesis\Data\Skyrim Special Edition\KnowYourArmorPatcher" --DefaultDataFolderPath C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Git\pgzadd3g.e31\Runner\KnowYourArmorPatcher\Data --GameRelease SkyrimSE --LoadOrderFilePath C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Workspace\Plugins.txt --OutputPath "C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Workspace\1 - KnowYourArmorPatcher" --PatcherName KnowYourArmorPatcher --PersistencePath C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Persistence
Prepping state.
Using TextFileSharedFormKeyAllocator allocator
Reading settings: C:\Games\Phoenix Flavour\tools\Synthesis\Data\Skyrim Special Edition\KnowYourArmorPatcher\settings.json
Settings Used: 
effect_intensity: 1.05

System.InvalidCastException: Unable to cast object of type 'Newtonsoft.Json.Linq.JObject' to type 'Newtonsoft.Json.Linq.JArray'.
   at KnowYourArmorPatcher.Program.RunPatch(IPatcherState`2 state) in C:\Users\Administrator\AppData\Local\Temp\Synthesis\woq0jem3.so5\Git\pgzadd3g.e31\Runner\KnowYourArmorPatcher\Program.cs:line 325
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.<>c__DisplayClass12_0`2.<<AddPatch>b__0>d.MoveNext() in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\SynthesisPipeline.cs:line 101
--- End of stack trace from previous location ---
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.Run(RunSynthesisMutagenPatcher args, ModKey exportKey, IFileSystem fileSystem) in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\SynthesisPipeline.cs:line 507
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.Run(RunSynthesisMutagenPatcher args, IFileSystem fileSystem) in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\SynthesisPipeline.cs:line 428
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.<>c__DisplayClass36_0.<<InternalRun>b__1>d.MoveNext() in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\SynthesisPipeline.cs:line 400
Synthesis.Bethesda.Execution.CliUnsuccessfulRunException: Error running solution patcher
   at Synthesis.Bethesda.Execution.Patchers.Running.Solution.SolutionPatcherRunner.Run(RunSynthesisPatcher settings, CancellationToken cancel)
   at Synthesis.Bethesda.Execution.Patchers.Running.Solution.SolutionPatcherRun.Run(RunSynthesisPatcher settings, CancellationToken cancel)
   at Synthesis.Bethesda.Execution.Running.Runner.RunAPatcher.Run(ModKey outputKey, PatcherPrepBundle prepBundle, CancellationToken cancellation, Nullable`1 sourcePath, String persistencePath)

</details>

**Comment by Heartbreaker714** [comment]
I think I figured it out. Solution: I had to not only delete the file in the AppData\Local\Temp folder, but I also had to delete the settings file in the Synthesis folder. This needs to be done in this order to work:

1. Delete the KnowYourArmor patcher in Synthesis
2. Close Synthesis
3. Delete the "..._AppData\Local\Temp\Synthesis\_"
4. Delete the KnowYourArmor folder located in the Synthesis program folder (My settings were located in _\tools\Synthesis\Data\Skyrim Special_ Edition\_KnowYourArmorPatcher_)
5. Restart Synthesis
6. Download the KnowYourArmor patcher
7. Run the patcher

After doing things in this order it worked perfectly
